streams/src/test/java/org/apache/kafka/test/MockProcessor.java

```diff
@@ -28,9 +29,11 @@
 import java.util.List;
 import java.util.Map;
 
-@SuppressWarnings("deprecation") // Old PAPI. Needs to be migrated.
-public class MockProcessor<K, V> extends org.apache.kafka.streams.processor.AbstractProcessor<K, V> {
-    private final MockApiProcessor<K, V, Object, Object> delegate;
+public class MockProcessor<KIn, VIn, KOut, VOut> implements Processor<KIn, VIn, KOut, VOut> {
+    private final MockApiProcessor<KIn, VIn, KOut, VOut> delegate;
+
+    private ProcessorContext<KOut, VOut> context;
```

Comment:
Why do we add this? We inherit `context` from the super class anyway.

Reply:
yup, should use the delegate. Will change.